### PR TITLE
Parse show notes

### PR DESCRIPTION
**Pull request by BasThomas**
This adds a format for show notes that will parse them to HTML.

The following format is expected:

```
# title
- description1: url1
- description2: url2

- description3: url3

# anotherTitle
- description4: url4
```

This will generate:

```html
<div>
  <p>
  <h3>title</h3>
  <ul>
    <li><a href="url1">description1</a></li>
    <li><a href="url2">description2</a></li>
  </ul>
  <ul>
    <li><a href="url3">description3</a></li>
  </ul>
  </p>
  <p>
  <h3>anotherTitle</h3>
  <ul>
    <li><a href="url4">description4</a></li>
  </ul>
  </p>
<div>
```

### CODE REVIEW

**Comment by terhechte** [comment]
What happened to this? It still says [WIP]. Is it ready for review? :)

**Comment by BasThomas** [comment]
Yeah not super happy with the code. It works in theory, but want to take another look at making this a bit less error prone / easier to expand on later

File: Sources/Masse/ConfigEntryParser.swift [hunk 1]
```diff
@@ -21,9 +21,15 @@ struct ConfigEntryParser {
         var hasReachedSeparator = false
         var result: [String: String] = [:]
         var noteLines: [String] = []
+        var pos = 0
```
Comment (BasThomas) [inline Sources/Masse/ConfigEntryParser.swift]: The position of what? I think that could help prevent some questions later 😄

File: Sources/Masse/ConfigEntryParser.swift [hunk 2]
```diff
@@ -21,9 +21,15 @@ struct ConfigEntryParser {
...
         for line in contents.components(separatedBy: .newlines) {
             if hasReachedSeparator {
                 noteLines.append(line)
+                let remainder = String(contents[contents.index(contents.startIndex, offsetBy: pos)..<contents.endIndex])
```
Comment (BasThomas) [inline Sources/Masse/ConfigEntryParser.swift]: I think you wrote an extension for this, no?

File: Sources/Masse/MiniMarkdownParser.swift [hunk 3]
```diff
@@ -0,0 +1,166 @@
+import Foundation
+
+/// Split into lines by `\n` however keep empty lines as empty strings
+internal func splitWithBreaks(lines: String) -> [Substring] {
+    var results: [Substring] = []
+    var (startIndex, currentIndex) = (lines.startIndex, lines.startIndex)
```
Comment (BasThomas) [inline Sources/Masse/MiniMarkdownParser.swift]: Should we make `startIndex` a `let`, so it isn't unintentionally changed?

File: Sources/Masse/MiniMarkdownParser.swift [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    if startIndex < lines.endIndex {
+        results.append(lines[startIndex..<lines.endIndex])
+    }
+    return results
+}
+
+internal struct MiniMarkdownParser {
+    internal enum Element {
```
Comment (BasThomas) [inline Sources/Masse/MiniMarkdownParser.swift:4]: Ooh I like this :)

File: Sources/Masse/MiniMarkdownParser.swift [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+            case .headline(level: let level, title: let title):
+                buffer.append("<h\(level)>\(convertRefs(title))</h\(level)>")
+            case .footnote(identifier: let identifier, contents: let contents):
+                buffer.append("<div><strong><a name='\(identifier)'>[\(identifier)]:</a></strong> \(contents)</div>")
+            }
+            return buffer
+        }
+
+        internal func convertRefs(_ input: Substring) -> String {
+            enum RefType { case image, link }
```
Comment (BasThomas) [inline Sources/Masse/MiniMarkdownParser.swift:32]: Ooh, good you thought about `image`.

File: Tests/MasseTests/MiniMarkdownTests.swift [hunk 6]
```diff
@@ -0,0 +1,177 @@
+import XCTest
+@testable import Masse
+
+import Foundation
+
+final class MinimarkdownTests: XCTestCase {
```
Comment (BasThomas) [inline Tests/MasseTests/MiniMarkdownTests.swift:6]: Oh man, this looks neat! 😀

File: Tests/MasseTests/MiniMarkdownTests.swift [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+- List 2
+- list 3
+
+### Third headline
+
+longer pargraph
+
+[^1] Footnote 1. Done.
+"""
+        print(MiniMarkdownParser().parseHTML(content))
```
Comment (BasThomas) [inline Tests/MasseTests/MiniMarkdownTests.swift]: This is not being tested yet?

**Review by BasThomas — COMMENTED** [review]
So cool! Hope to get some time during the weekend to take another look and play with it. Thanks @terhechte!

**Pull request merged**